install/installation/installation.go

```diff
@@ -310,46 +322,53 @@ func (k KymaInstaller) installTiller(tillerYaml string, createFunction func([]k8
 	return nil
 }
 
-func (k KymaInstaller) deployInstallerForIstallation(installerYaml string) error {
-	return k.deployInstaller(installerYaml, k.k8sGenericClient.CreateResources)
+func (k KymaInstaller) deployInstallerForIstallation(yamlFile string) error {
+	return k.deployInstaller(yamlFile, k.k8sGenericClient.CreateResources)
 }
 
-func (k KymaInstaller) deployInstallerForUpgrade(installerYaml string) error {
-	return k.deployInstaller(installerYaml, k.k8sGenericClient.ApplyResources)
+func (k KymaInstaller) deployInstallerForUpgrade(yamlFile string) error {
+	return k.deployInstaller(yamlFile, k.k8sGenericClient.ApplyResources)
 }
 
-func (k KymaInstaller) deployInstaller(installerYaml string, createResourcesFunc func(resources []k8s.K8sObject) ([]*unstructured.Unstructured, error)) error {
+func (k KymaInstaller) deployInstaller(yamlFile string, createResourcesFunc func(resources []k8s.K8sObject) ([]*unstructured.Unstructured, error)) error {
 	k.infof("Deploying Installer...")
 
-	k8sInstallerObjects, err := k8s.ParseYamlToK8sObjects(k.decoder, installerYaml)
+	k8sInstallerObjects, err := k8s.ParseYamlToK8sObjects(k.decoder, yamlFile)
 	if err != nil {
-		return fmt.Errorf("failed to parse Installer yaml file to Kubernetes dynamicClientObjects: %w", err)
+		return fmt.Errorf("failed to parse yaml file to Kubernetes dynamicClientObjects: %w", err)
 	}
 
-	var installationCR *v1alpha1.Installation
-	installationCR, k8sInstallerObjects, err = k.extractInstallationCR(k8sInstallerObjects)
-	if err != nil {
-		return fmt.Errorf("failed to get Installation CR: %w", err)
-	}
-
-	_, found := installationCR.Labels[installationActionLabel]
-	if found {
-		delete(installationCR.Labels, installationActionLabel)
-	}
+	// Check if its InstallerCR or the InstallerYaml
+	if len(k8sInstallerObjects) == 1 {
+		if k8sInstallerObjects[0].GVK.Group == v1alpha1.Group && k8sInstallerObjects[0].GVK.Kind == "Installation" {
+			installationCR, ok := k8sInstallerObjects[0].Object.(*v1alpha1.Installation)
+			if !ok {
+				return fmt.Errorf("unexpected type of Installation object: %T, failed to cast to *Installation", k8sInstallerObjects[0].Object)
+			}
+			_, found := installationCR.Labels[installationActionLabel]
+			if found {
+				delete(installationCR.Labels, installationActionLabel)
+			}
+			k.infof("Applying Installation CR modifications...")
+			k.installationOptions.installationCRModificationFunc(installationCR)
+			k.infof("Applying Installation CR...")
+			err = k.applyInstallationCR(installationCR)
+			if err != nil {
+				return fmt.Errorf("failed to apply Installation resources: %w", err)
+			}
+			k.infof("Installer CR deployed.")
 
-	_, err = createResourcesFunc(k8sInstallerObjects)
```

Comment:
Here we apply all resources from installation artifacts besides Installation CR so the Installation CRD should be applied here.